### PR TITLE
[Network] MACsec Configuration Edits

### PR DESCRIPTION
**Pull request by francescanarea**
**Description**  
- Fix root cause of [Issue #7569](https://github.com/Azure/azure-rest-api-specs/issues/7569) by removing dashes from cipher values ([sister PR](https://github.com/Azure/azure-rest-api-specs/pull/10181))
- Add sci mode parameter to macsec configuration with Enabled/Disables

**Testing Guide**  
Needs to be tested.

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S173

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -2760,12 +2761,11 @@ def update_express_route_port_link(cmd, instance, express_route_port_name, link_
         instance.links[link_index].mac_sec_config.cak_secret_identifier = macsec_cak_secret_identifier
         instance.links[link_index].mac_sec_config.ckn_secret_identifier = macsec_ckn_secret_identifier
 
-        # TODO https://github.com/Azure/azure-rest-api-specs/issues/7569
-        # need to remove this conversion when the issue is fixed.
         if macsec_cipher is not None:
-            macsec_ciphers_tmp = {'gcm-aes-128': 'GcmAes128', 'gcm-aes-256': 'GcmAes256'}
-            macsec_cipher = macsec_ciphers_tmp[macsec_cipher]
-        instance.links[link_index].mac_sec_config.cipher = macsec_cipher
+            instance.links[link_index].mac_sec_config.cipher = macsec_cipher
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:2765]: Could you confirm that service is ready? Please add one test for this argument.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -2760,12 +2761,11 @@ def update_express_route_port_link(cmd, instance, express_route_port_name, link_
...
+
+        if macsec_sci_state is not None:
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:2767]: Please add one test for this argument as well. Thanks 😄

**Comment by MyronFanQiu** [comment]
Something wrong with github or ADO, please merge the latest dev branch into your branch to fix linter CI issue.

**Comment by MyronFanQiu** [comment]
@francescanarea Hi. Do we need this PR in this release train? Please solve the comments and I can help you merge this PR.

**Comment by MyronFanQiu** [comment]
@francescanarea Hi, do you still need this PR?

**Comment by haroldrandom** [comment]
Closed as inactive

**Pull request closed without merging**